src/v/transform/api.cc

```diff
@@ -230,9 +233,33 @@ service::service(
 
 service::~service() = default;
 
-ss::future<> service::start() { throw std::runtime_error("unimplemented"); }
+ss::future<> service::start() {
+    std::unique_ptr<partition_source_factory> source_factory
+      = std::make_unique<partition_source_factory>(
+        &_partition_manager->local());
+    std::unique_ptr<sink::factory> sink_factory
+      = std::make_unique<rpc_client_factory>(&_rpc_client->local());
+
+    _manager = std::make_unique<manager<ss::lowres_clock>>(
+      std::make_unique<registry_adapter>(
+        &_plugin_frontend->local(), &_partition_manager->local()),
+      std::make_unique<proc_factory>(
+        [this](model::transform_metadata meta) {
+            return create_engine(std::move(meta));
+        },
+        std::move(source_factory),
+        std::move(sink_factory)));
+    co_await _manager->start();
+}
 
-ss::future<> service::stop() { throw std::runtime_error("unimplemented"); }
+ss::future<> service::stop() {
+    co_await _gate.close();
+    // It's possible to call stop before start, so make sure we created the
+    // manager.
```

Comment:
but ::stop is still a terminal state for the instance if it is called?

Reply:
Yes, should I add this to the comment? If startup fails after the transform service is created this can happen, because we always run the stop methods of things that were created (and not started). 

I ran into this a while ago when I had a bug that stopped startup and stop was called which assumed there was always a manager.

Reply:
>Yes, should I add this to the comment?
i don't think it's necessary. i think it's a good property for a service to have anyway